### PR TITLE
Endpoint for dashboard to retrieve environment specific settings

### PR DESCRIPTION
**Pull request by oudeismetis**
Related to istrategylabs/franklin#95

### CODE REVIEW

**Comment by camiriveron** [comment]
:+1:

File: franklin/core/views.py [hunk 1]
```diff
@@ -10,6 +11,22 @@
...
+    For the dashboard to initialize itself with the api
+    ---
+    type:
+        client_id:
+            type: string
+            description: github client_id needed for social auth
+            required: true
+    """
+    message = {'client_id': os.environ['SOCIAL_AUTH_GITHUB_KEY']}
+    return Response(message, status=status.HTTP_200_OK)
```
Comment (oudeismetis) [inline franklin/core/views.py]: For now we aren't going to secure this endpoint. Any data this returns would be readable by inspecting the FE dashboard anyway.

File: franklin/core/helpers.py [hunk 2]
```diff
@@ -104,7 +104,11 @@ def authenticate(self, request):
             backend = 'github'
             backend = load_backend(strategy, backend,
                                    reverse(path, args=(backend,)))
-            user = backend.do_auth(access_token=oauth_token)
+            try:
+                user = backend.do_auth(access_token=oauth_token)
+            except requests.HTTPError as e:
+                msg = e.response.json()
+                raise exceptions.AuthenticationFailed(msg)
```
Comment (oudeismetis) [inline franklin/core/helpers.py]: got this error when I was testing @camiriveron github token on my local franklin, which uses a different github application. So yea, I can confirm that you can only auth with tokens for that environments app.

**Comment by JulianGindi** [comment]
:+1:

**Comment by oudeismetis** [comment]
merged

**Pull request closed without merging**